File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int l = 0;
        int r = k-1;

        int ans = INT_MAX;

        while(r < nums.size()){
            ans = min(ans, nums[r]-nums[l]);
            l+=1;
            r +=1;
        }

        return ans;

        
    }
};

File: 2112-minimum-difference-between-highest-and-lowest-of-k-scores/2112-minimum-difference-between-highest-and-lowest-of-k-scores.swift
class Solution {
    func minimumDifference(_ nums: [Int], _ k: Int) -> Int {

       var ans = Int.max
       var num = nums.sorted()
       var l = 0
       var r = k-1

       while r < nums.count {
        ans = min(ans, num[r]-num[l])
        l+=1
        r+=1
       }

       return ans

        
    }
}